src/rpc/mining.cpp

```diff
@@ -496,7 +496,7 @@ static RPCHelpMan prioritisetransaction()
 
     // Non-0 fee dust transactions are not allowed for entry, and modification not allowed afterwards
     const auto& tx = mempool.get(hash);
-    if (tx && HasDust(tx, mempool.m_opts.dust_relay_feerate)) {
+    if (mempool.m_opts.require_standard && tx && !GetDust(*tx, mempool.m_opts.dust_relay_feerate).empty()) {
```

Comment:
Setting `-dustrelayfee=0` avoids the check much more directly and simply, with no recompilation needed.

Gating it based on standardness checks is just easier to understand when e.g., standardness isn't being enforced anyways, so dust should be treated as a no-op.

I'm going to mark this as resolved. If we want to reverse this logic, imo we should do it in another PR that removes both the logic in prioritisation as well as the modified fee checks.

Reply:
fwiw I'm not too concerned about this "encouraging" miners to patch. Miners might mine dust and this doesn't stop them from doing anything, but I don't think this would make that more likely given "subverting" it would only require `-dustrelayfee=0` and has always been possible. I think it's worth discussing and asking miners. But that's my thinking.